shared/search/conversion.ts

```diff
@@ -1,25 +1,42 @@
 import * as sourcegraph from 'sourcegraph'
-import { Result } from '../language-specs/languages'
 import { LineMatch, SearchResult, SearchSymbol } from '../util/api'
 import { isDefined } from '../util/util'
 
 /**
- * Convert an internal result into a Sourcegraph location.
- *
- * @param result The search result.
+ * The exploded version of a search result. Each symbol and indexed/un-indexed
+ * search result becomes a single instance of this interface.
  */
-export function resultToLocation(result: Result): sourcegraph.Location {
-    return {
-        uri: new URL(
-            `git://${result.repo}?${result.rev || 'HEAD'}#${result.file}`
-        ),
-        range: new sourcegraph.Range(
-            result.start.line,
-            result.start.character,
-            result.end.line,
-            result.end.character
-        ),
-    }
+export interface Result {
+    /**
```

Comment:
As above, can we compact these one-liners a bit?

```suggestion
/** The name of the repository containing the result. */
```

et seq.